### PR TITLE
parametrize a whole mess of tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
What it says on the tin.  Because this is a big diff, nothing is moved or changed other than parametrizing tests.

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_datetime.py [hunk 1]
```diff
@@ -381,18 +380,17 @@ def test_ns_index(self):
                                      freq=index.freq)
         self.assert_index_parameters(new_index)
 
-    def test_join_with_period_index(self):
+    @pytest.mark.parametrize('how', ['left', 'right', 'inner', 'outer'])
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: note to make a fixture and hence conform ordering of how

File: pandas/tests/indexes/datetimes/test_datetime.py [hunk 2]
```diff
@@ -439,18 +437,18 @@ def test_factorize(self):
         tm.assert_numpy_array_equal(arr, exp_arr)
         tm.assert_index_equal(idx, idx3)
 
-    def test_factorize_tz(self):
-        # GH 13750
-        for tz in [None, 'UTC', 'US/Eastern', 'Asia/Tokyo']:
-            base = pd.date_range('2016-11-05', freq='H', periods=100, tz=tz)
-            idx = base.repeat(5)
+    @pytest.mark.parametrize('tz', [None, 'UTC', 'US/Eastern', 'Asia/Tokyo'])
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_datetime.py]: usual note to fixturize tz (in future)

File: pandas/tests/indexes/period/test_arithmetic.py [hunk 3]
```diff
@@ -613,42 +613,42 @@ def test_pi_ops(self):
...
-    def test_pi_ops_errors(self):
+    @pytest.mark.parametrize('ng', ["str", 1.5])
+    def test_pi_ops_errors(self, ng):
         idx = PeriodIndex(['2011-01', '2011-02', '2011-03', '2011-04'],
                           freq='M', name='idx')
         ser = pd.Series(idx)
 
         msg = r"unsupported operand type\(s\)"
 
         for obj in [idx, ser]:
```
Comment (jreback) [inline pandas/tests/indexes/period/test_arithmetic.py]: still need this loop here?

File: pandas/tests/scalar/period/test_period.py [hunk 4]
```diff
@@ -212,58 +212,59 @@ def test_period_cons_combined(self):
...
-            p = Period('1/1/2005', freq='A').to_timestamp(freq='3H', tz=case)
-            exp = Timestamp('1/1/2005', tz='UTC').tz_convert(case)
-            exp_zone = pytz.timezone(case).normalize(p)
-
-            assert p == exp
-            assert p.tz == exp_zone.tzinfo
-            assert p.tz == exp.tz
-
-    def test_timestamp_tz_arg_dateutil(self):
+    @pytest.mark.parametrize('case', ['Europe/Brussels',
```
Comment (jreback) [inline pandas/tests/scalar/period/test_period.py]: rename to tz

File: pandas/tests/scalar/period/test_period.py [hunk 5]
```diff
@@ -212,58 +212,59 @@ def test_period_cons_combined(self):
...
+
+        p = Period('1/1/2005', freq='A').to_timestamp(freq='3H', tz=case)
+        exp = Timestamp('1/1/2005', tz='UTC').tz_convert(case)
+        exp_zone = pytz.timezone(case).normalize(p)
+
+        assert p == exp
+        assert p.tz == exp_zone.tzinfo
+        assert p.tz == exp.tz
+
+    @pytest.mark.parametrize('case', ['dateutil/Europe/Brussels',
```
Comment (jreback) [inline pandas/tests/scalar/period/test_period.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, just a couple of comments

File: pandas/tests/indexes/period/test_arithmetic.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/period/test_arithmetic.py]: There are a bunch of borderline cases like this left for the next pass.

File: pandas/tests/scalar/period/test_period.py (hunk 5 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/period/test_period.py]: There's already a `tz` variable in the method; `case` is the name used for this in the status quo.

Comment (jreback) [reply pandas/tests/scalar/period/test_period.py]: pls change. this is really confusing.

File: pandas/tests/indexes/period/test_arithmetic.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/indexes/period/test_arithmetic.py]: thats fine

**Review by jreback — APPROVED** [review]

**Pull request merged**